src/index.ts

```diff
@@ -1,26 +1,42 @@
-import { ReplaySubject } from 'rxjs'
+import { Observable, ReplaySubject } from 'rxjs'
 
-type SubjectByProp = Map<string, ReplaySubject<any>>
+interface Cache { observable: Observable<any>; subject: ReplaySubject<any> }
+type CacheByProp = Map<string, Cache>
 
-const subjects: WeakMap<Object, SubjectByProp> = new WeakMap()
+const cache: WeakMap<Object, CacheByProp> = new WeakMap()
```

Comment:
I believe you can do: `const cache = new WeakMap<Object, CacheByProp>()` since about typescript 2.2.